server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -342,48 +355,63 @@ public void listenForUpdates(ClusterSettings clusterSettings) {
     }
 
     private synchronized void updateSkipUnavailable(String clusterAlias, Boolean skipUnavailable) {
-        RemoteClusterConnection remote = this.remoteClusters.get(clusterAlias);
+        RemoteClusterConnection remote = getConnectionsMapForCurrentProject().get(clusterAlias);
         if (remote != null) {
             remote.setSkipUnavailable(skipUnavailable);
         }
     }
 
+    @FixForMultiProject(description = "Refactor as needed to support project specific changes to linked remotes.")
     public synchronized void updateRemoteClusterCredentials(Supplier<Settings> settingsSupplier, ActionListener<Void> listener) {
+        final var projectId = projectResolver.getProjectId();
         final Settings settings = settingsSupplier.get();
         final UpdateRemoteClusterCredentialsResult result = remoteClusterCredentialsManager.updateClusterCredentials(settings);
         // We only need to rebuild connections when a credential was newly added or removed for a cluster alias, not if the credential
         // value was updated. Therefore, only consider added or removed aliases
         final int totalConnectionsToRebuild = result.addedClusterAliases().size() + result.removedClusterAliases().size();
         if (totalConnectionsToRebuild == 0) {
-            logger.debug("no connection rebuilding required after credentials update");
+            logger.debug("project ID [{}] no connection rebuilding required after credentials update", projectId);
```

Comment:
I think this `ID` is a leftover
```suggestion
            logger.debug("project [{}] no connection rebuilding required after credentials update", projectId);
```